components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/util/LightFileImportComponent.java

```diff
@@ -0,0 +1,1154 @@
+     * omero.gateway.model.DatasetData, java.lang.Object)
+     */
+    @Override
+    public void setLocation(DataObject data, DatasetData dataset, Object refNode) {
+        this.data = data;
+        this.dataset = dataset;
+        this.refNode = refNode;
+        if (refNode != null && refNode instanceof TreeImageDisplay) {
+            TreeImageDisplay n = (TreeImageDisplay) refNode;
+            Object ho = n.getUserObject();
+            if (ho instanceof DatasetData || ho instanceof ProjectData
+                    || ho instanceof ScreenData) {
+                containerObject = (DataObject) ho;
+            }
+            return;
+        }
+        if (dataset != null) {
+            containerObject = dataset;
+            return;
+        }
+        if (data != null && data instanceof ScreenData) {
+            containerObject = data;
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #setImportLogFile(java.util.Collection, long)
+     */
+    @Override
+    public void setImportLogFile(Collection<FileAnnotationData> data, long id) {
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #getDataset()
+     */
+    @Override
+    public DatasetData getDataset() {
+        return dataset;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #getDataObject()
+     */
+    @Override
+    public DataObject getDataObject() {
+        return data;
+    }
+
+    /**
+     * Replaces the initial status label.
+     * 
+     * @param label
+     *            The value to replace.
+     */
+    void setStatusLabel(Status label) {
+        status = label;
+        status.addPropertyChangeListener(this);
+    }
+
+    /**
+     * Sets the parent of the component.
+     * 
+     * @param parent
+     *            The value to set.
+     */
+    void setParent(FileImportComponentI parent) {
+        this.parent = parent;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #hasParent()
+     */
+    @Override
+    public boolean hasParent() {
+        return parent != null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #getStatus()
+     */
+    @Override
+    public Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Returns the associated file if any.
+     *
+     * @param series
+     *            See above.
+     * @return See above.
+     */
+    private FileObject getAssociatedFile(int series) {
+        List<FileObject> l = getFile().getAssociatedFiles();
+        Iterator<FileObject> i = l.iterator();
```

Comment:
seems to be a long way to write,
```java
for (FileObject f : getFile().getAssociatedFiles()) {
    if (f.getIndex() == series) {
    ...
```
Fine to use Java 5 in new code. (Though nothing uses this method anyway.)